tools/screenshot-test/src/app/firebase.service.ts

```diff
@@ -0,0 +1,204 @@
+        let childValue = childSnapshot.val();
+        switch (childSnapshot.key) {
+          case 'sha':
+            this._processSha(childValue);
+            break;
+          case 'travis':
+            this.screenshotResult.travis = childValue;
+            break;
+          case 'results':
+            this._processResults(childSnapshot);
+            break;
+          case 'approved':
+            this.screenshotResult.approved = childValue;
+            break;
+        }
+        counter ++;
+        if (counter == snapshot.numChildren()) {
+          return true;
+        }
+      });
+    });
+  }
+
+  signInGithub(): firebase.Promise<any> {
+    return firebase.auth().signInWithRedirect(new firebase.auth.GithubAuthProvider());
+  }
+
+  signOutGithub() {
+    firebase.auth().signOut();
+  }
+
+  /** Change the PR status to approved to let Firebase Functions copy test images to goldens */
+  approvePR() {
+    return this._databaseRef().child('approved').set(`${this.screenshotResult.sha}-${new Date()}`);
+  }
+
+  /** Change the commit's screenshot test status on GitHub */
+  updatePRResult() {
+    return this._databaseRef().child('result').child(this.screenshotResult.sha).set(true);
+  }
+
+  _databaseRef(): firebase.database.Reference {
+    return firebase.database().ref('screenshot').child('reports')
+      .child(this.screenshotResult.prNumber);
+  }
+
+  _storageRef(): firebase.storage.Reference {
+    return firebase.storage().ref('screenshots').child(this.screenshotResult.prNumber);
+  }
+
+  /** Put the results in screenshotReuslt */
+  _processResults(childSnapshot: firebase.database.DataSnapshot) {
+    let childCounter = 0;
+    this.screenshotResult.collapse = [];
+    this.screenshotResult.testnames = [];
+    this.screenshotResult.results.clear();
+    childSnapshot.forEach((resultSnapshot: firebase.database.DataSnapshot) => {
+      let key = resultSnapshot.key;
+      let value = resultSnapshot.val();
+      this.screenshotResult.results.set(key, value);
+      this.screenshotResult.testnames.push(key);
+      this.screenshotResult.collapse.push(value);
+      childCounter++;
+      if (childCounter == childSnapshot.numChildren()) {
+        return true;
+      }
+    });
+  }
+
+  _processSha(childValue) {
+    this.screenshotResult.sha = childValue;
+    // Get github status
+    this.getGithubStatus().then((result) => this.screenshotResult.githubStatus = result);
+    // Get test result
+    this._databaseRef().child(`result/${childValue}`).once('value')
+      .then((dataSnapshot: firebase.database.DataSnapshot) => {
+        this.screenshotResult.result = dataSnapshot.val();
+      });
+  }
+
+  getGithubStatus() {
+    let url =
+      `https://api.github.com/repos/${config.repoSlug}/commits/${this.screenshotResult.sha}/status`;
+    return new Promise((resolve) => {
+      request({
+        url: url,
+        method: 'GET',
+      }, function (error: any, response: any) {
```

Comment:
Use arrow function? And probably move the `{ url, method: 'GET' }` into a single line?